packages/plugins/i18n/admin/src/components/CMHeaderActions.tsx

```diff
@@ -50,6 +50,59 @@ import type { I18nBaseQuery } from '../types';
  * LocalePickerAction
  * -----------------------------------------------------------------------------------------------*/
 
+interface LocaleOptionProps {
+  isDraftAndPublishEnabled: boolean;
+  locale: Locale;
+  status: 'draft' | 'published' | 'modified';
+  entryExists: boolean;
+}
+
+const LocaleOption = ({
+  isDraftAndPublishEnabled,
+  locale,
+  status,
+  entryExists,
+}: LocaleOptionProps) => {
+  const { formatMessage } = useIntl();
+
+  if (!entryExists) {
+    return formatMessage(
+      {
+        id: getTranslation('CMEditViewLocalePicker.locale.create'),
+        defaultMessage: 'Create <bold>{locale}</bold> locale',
+      },
+      {
+        bold: (locale: React.ReactNode) => <b>{locale}</b>,
+        locale: locale.name,
+      }
+    );
+  }
+
+  return (
+    <Flex width="100%" gap={1} justifyContent="space-between">
+      <Typography>{locale.name}</Typography>
+      {isDraftAndPublishEnabled ? (
+        <Status
+          display="flex"
+          paddingLeft="6px"
+          paddingRight="6px"
+          paddingTop="2px"
+          paddingBottom="2px"
+          showBullet={false}
+          size={'S'}
+          variant={
+            status === 'draft' ? 'secondary' : status === 'published' ? 'success' : 'alternative'
```

Comment:
I find nested ternaries quite hard to read. I think you could try either using a simple `getVariantColor` function, or using an object to map each status to a color and do `statusColors[status]`